source/kraken/apply_disruption.cpp

```diff
@@ -520,15 +520,31 @@ void apply_impact(boost::shared_ptr<nt::disruption::Impact> impact,
     LOG4CPLUS_TRACE(log4cplus::Logger::getInstance("log"), impact->uri << " impact added");
 }
 
+using impact_wptr = boost::weak_ptr<nt::disruption::Impact>;
+
+static auto comp = [](const impact_wptr& lhs, const impact_wptr& rhs){
+    if(lhs.expired() || rhs.expired()) {
+        return false;
+    }
+    auto i_lhs = lhs.lock();
+    auto i_rhs = rhs.lock();
```

Comment:
I checked that at L526... Isn't that enough? (Hopefully, if weak_ptr is not expired, the resulting shared_ptr is not null neither)